crates/oxc_linter/src/rules/eslint/no_return_await.rs

```diff
@@ -0,0 +1,255 @@
+                }
+            },
+            AstKind::LogicalExpression(logic_expr) => {
+                if logic_expr.right.span() == node.kind().span() {
+                    return is_in_tail_call_position(parent, ctx);
+                }
+            },
+            AstKind::SequenceExpression(seq_expr) => {
+                if let Some(seq_expr_last) = seq_expr.expressions.last() 
+                    && seq_expr_last.span() == node.kind().span(){
+                        return is_in_tail_call_position(parent, ctx);
+                }
+            },
+            // `return (await b())`
+            AstKind::ParenthesizedExpression(paren_expr) => {                    
+                if paren_expr.expression.span() == node.kind().span() {
+                    return is_in_tail_call_position(parent, ctx);
+                }
+            },
+            // async () => await bar()
+            AstKind::ExpressionStatement(expr_stat) => {
+                // expression state in the last line of a `function_body`
+                if expr_stat.expression.span() == node.kind().span() {
+                    if let Some(grand_parent) = ctx.nodes().parent_node(parent.id()) {
+                        if let AstKind::FunctionBody(func_body) = grand_parent.kind() {
+                            if let Some(func_body_stat_last) = func_body.statements.last() {
+                                return func_body_stat_last.span() == node.kind().span();
+                            }
+                        }
+                    }
+                }
+            },
+            _ => {
+                return false;
+            }
+        }
+    }
+    false
+}
+
+fn has_error_handler<'a>(node: &AstNode<'a>, ctx: &LintContext<'a>) -> bool {
+    let mut current_node = node;
+    loop {
+        if let Some(parent_node) = ctx.nodes().parent_node(current_node.id()) {
+            let parent_node_kind = parent_node.kind();
+            if matches!(parent_node_kind, AstKind::Program(_)) {
+                break;
+            }
+
+            if parent_node_kind.is_function_like() {
+                break;
+            }
+
+            if let AstKind::TryStatement(try_stat) = parent_node_kind {
+                let current_node_span = current_node.kind().span();
+                // try statement must have a `catch` or `finally`
+                if try_stat.block.span == current_node_span {
+                    return true;
+                }
+
+                // return await in `catch clause` with `finally` would be passed
+                if let Some(catch_clause) = &try_stat.handler {
+                    if catch_clause.span == current_node_span && try_stat.finalizer.is_some() {
+                        return true;
+                    }
+                }
+            }
+
+            current_node = parent_node;
+        }
+    }
+    false
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
```

Comment:
Try running `just new-rule no-return-await` and copy the tests over, so we can get rid of `\n` and `\t`

Reply:
I found that we would lose the last 8 test cases yesterday.
<img width="633" alt="image" src="https://github.com/Boshen/oxc/assets/6828924/c763c03a-a472-4766-a7fa-b15b8ab23918">

Reply:
Updated, but the last 8 test cases are copied by hand